iOS/DuckDuckGo/SettingsSubscriptionView.swift

```diff
@@ -264,28 +264,28 @@ struct SettingsSubscriptionView: View {
 
         if subscriptionFeatures.contains(.dataBrokerProtection) {
             let hasDBPEntitlement = userEntitlements.contains(.dataBrokerProtection)
-
-            if DataBrokerProtectionIOSManager.isDBPStaticallyEnabled {
-                NavigationLink(destination: LazyView(DataBrokerProtectionViewControllerRepresentation(dbpViewControllerProvider: DataBrokerProtectionIOSManager.shared!)), isActive: $isShowingDBP) {
-                    SettingsCellView(
-                        label: UserText.settingsPProDBPTitle,
-                        image: Image(uiImage: DesignSystemImages.Color.Size24.identity),
-                        statusIndicator: StatusIndicatorView(status: hasDBPEntitlement ? .on : .off),
-                        isGreyedOut: !hasDBPEntitlement
-                    )
-                }
-                .disabled(!hasDBPEntitlement)
-            } else {
-                NavigationLink(destination: LazyView(SubscriptionPIRMoveToDesktopView()), isActive: $isShowingDBP) {
-                    SettingsCellView(
-                        label: UserText.settingsPProDBPTitle,
-                        image: Image(uiImage: DesignSystemImages.Color.Size24.identity),
-                        statusIndicator: StatusIndicatorView(status: hasDBPEntitlement ? .on : .off),
-                        isGreyedOut: !hasDBPEntitlement
-                    )
+            let hasValidStoredProfile = settingsViewModel.dataBrokerProtectionIOSManager
+                .flatMap { try? $0.meetsProfileRunPrequisite } ?? false
+            let statusIndicator: StatusIndicator = hasDBPEntitlement && hasValidStoredProfile ? .on : .off
+
+            let destination: LazyView<AnyView> = {
+                if let dbpManager = settingsViewModel.dataBrokerProtectionIOSManager,
+                   DataBrokerProtectionIOSManager.isDBPStaticallyEnabled {
```

Comment:
Theoretically, successfully unwrapping `settingsViewModel.dataBrokerProtectionIOSManager ` should be enough, no need to check if it’s statically enabled. But I thought it would be nice to keep that check here.

Reply:
Cool, I'd definitely prefer we have the check here as well since I'd rather we didn't hardcode any assumptions about if dbpIOSManager exists or not. And the settings page shouldn't change if isDBPStaticallyEnabled is false